jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+from traitlets import observe
+
+from jdaviz.core.events import ViewerAddedMessage
+from jdaviz.core.marks import MarkersMark
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin, TableMixin
+from jdaviz.core.user_api import PluginUserApi
+
+__all__ = ['Markers']
+
+_default_table_values = {'x': np.nan,
+                         'x:unit': '',
+                         'y': np.nan,
+                         'y:unit': '',
+                         'xy:unreliable': False,
+                         'slice index': np.nan,
+                         'slice wavelength': np.nan,
+                         'slice wavelength:unit': '',
+                         'value': np.nan,
+                         'value:unit': '',
+                         'RA (ICRS)': '',
+                         'DEC (ICRS)': '',
+                         'RA (deg)': np.nan,
+                         'DEC (deg)': np.nan,
+                         'radec:unreliable': False,
+                         'index': np.nan}
+
+
+@tray_registry('g-markers', label="Markers")
+class Markers(PluginTemplateMixin, ViewerSelectMixin, TableMixin):
+    """
+    See the :ref:`Markers Plugin Documentation <markers-plugin>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * :meth:`clear_table`
+    * :meth:`~jdaviz.core.template_mixin.TableMixin.export_table`
+    """
+    template_file = __file__, "markers.vue"
+
+    @property
+    def user_api(self):
+        return PluginUserApi(self, expose=('clear_table', 'export_table',))
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        headers = ['x', 'x:unit', 'y', 'y:unit']
+
+        if self.config == 'cubeviz':
+            headers += ['slice index', 'slice wavelength', 'slice wavelength:unit']
+
+        if self.config in ['imviz', 'cubeviz', 'mosviz']:
+            # image viewers
+            headers += ['xy:unreliable',
+                        'RA (ICRS)', 'DEC (ICRS)',
+                        'RA (deg)', 'DEC (deg)',
+                        'radec:unreliable',
+                        'value', 'value:unit',
+                        'viewer']
+        if self.config in ['specviz', 'specviz2d', 'mosviz']:
+            # 1d spectrum viewers
+            headers += ['index']
+        if self.config in ['specviz2d', 'mosviz']:
+            # 2d spectrum viewers
+            headers += []

```

Comment:
I don't see the point of adding empty list here, or did I misunderstand this code?
```suggestion
        # NOTE: Uncomment when there is something to add.
        # if self.config in ('specviz2d', 'mosviz'):
        #     # 2d spectrum viewers
        #     headers += []
```